generators/app/templates/shared/jenkins-seed.groovy

```diff
@@ -0,0 +1,115 @@
+import javaposse.jobdsl.dsl.helpers.step.StepContext
+import javaposse.jobdsl.dsl.Context
+import javaposse.jobdsl.dsl.helpers.*
+
+class DockerisedService {
+
+    String name
+    String gitRepoUrl
+    String dockerRepoUrl
+    Boolean requiresMysql
+
+    public DockerisedService(name, gitRepoUrl, dockerRepoUrl, requiresMysql) {
+        this.name = name
+        this.gitRepoUrl = gitRepoUrl
+        this.requiresMysql = requiresMysql
+        this.dockerRepoUrl = dockerRepoUrl
+    }
+}
+
+DockerisedService dockerisedService = new DockerisedService(
+    <%= props.projectName %>, <%= props.gitRepoUrl %>,
+    <%= props.dockerRepoUrl %>, <%= props.testsRequireMysql %>
+)
+
+/**
+ *
+ * Creates the job in jenkins
+ *
+ * @param isReleaseJob If true it will build and push the docker image
+ *
+ */
+def createJob(service, isReleaseJob) {
+    jobName = "${service-name}-ci"
+    if (isReleaseJob) jobName += "-release"
+
+    job(jobName) {
+        logRotator(35,35)
+
+        label('build-nodes')
+
+        parameters {
+            stringParam('GIT_BRANCH', 'master', 'Can be SHA, Tag or Branch')
+            stringParam('DOCKER_IMAGE_TAG', 'latest', 'The docker image will be tagged using this tag')
+        }
+
+        scm {
+            git {
+                remote {
+                    url(service.gitRepoUrl)
+                }
+                branch("$\GIT_BRANCH")
+            }
+        }
+
+        concurrentBuild(true)
+
+        wrappers {
+            buildName('#${BUILD_NUMBER} - ${ENV,var="GitBranch"}')
+        }
+
+        steps {
+            if (service.requiresMysql) startMysqlStep()
+
+            if (isReleaseJob) {
+                sbtCommandStep('release')
+                dockerBuildAndPushStep(service)
+            } else {
+                sbtCommandStep('test')
```

Comment:
Should this run sbt stage as well? The release command will run stage (or we can run stage ourselves in here) but doing a test will compile so I can't see a stage failing if a test doesn't.